src/app/components/book-form/book-form.component.ts

```diff
@@ -1,27 +1,86 @@
-import { Component, OnInit, ChangeDetectionStrategy } from '@angular/core';
-import { FormBuilder, FormGroup } from '@angular/forms';
+import { Component, ChangeDetectionStrategy, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { FormGroup, FormBuilder, Validators, AbstractControl } from '@angular/forms';
+import { IReadBook } from '@app/shared/models/read-book.model';
+import { MatDialogRef } from '@angular/material/dialog';
+import { EditDialogComponent } from '../book-dialog/edit/edit-dialog.component';
+import { AddDialogComponent } from '../book-dialog/add/add-dialog.component';
+import { FormActions, BookFormFields } from '@app/shared/enums';
+import { AbstractSubscriptionComponent } from '@app/abstracts/abstract-subscription.component';
 
 @Component({
   selector: 'app-book-form',
   templateUrl: './book-form.component.html',
   styleUrls: ['./book-form.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class BookFormComponent implements OnInit {
+export class BookFormComponent extends AbstractSubscriptionComponent implements OnInit {
```

Comment:
Why do we need this extension here?

Reply:
AbstractSubscriptionComponent handles subscription logic (eg. unsubscribing onDestroy), just like it's name suggests.